pilot/cmd/pilot-agent/config.go

```diff
@@ -39,7 +39,11 @@ import (
 func constructProxyConfig(role *model.Proxy) (meshconfig.ProxyConfig, error) {
 	annotations, err := readPodAnnotations()
 	if err != nil {
-		log.Warnf("failed to read pod annotations: %v", err)
+		if os.IsNotExist(err) {
```

Comment:
When can this happen

Reply:
VMs, running locally, etc. The labels can be set through other means like environment variables as well, files are just there to support k8s